### PR TITLE
#9. TCC and LCC were introduced

### PR DESCRIPTION
**Pull request by skapral**
Fix for #9

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#108](https://github.com/yegor256/jpeek/issues/108) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#108](https://github.com/yegor256/jpeek/issues/108) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/main/resources/org/jpeek/metrics/LCC.xsl [hunk 1]
```diff
@@ -60,9 +60,9 @@ SOFTWARE.
     </xsl:variable>
     <!--
     @todo #9: `indirectly-related-pairs` are pairs of methods, which are through other directly connected methods.
```
Comment (amihaiemil) [inline src/main/resources/org/jpeek/metrics/LCC.xsl]: @skapral also add the estimation: ``@todo #9:30min [...]``. For the next puzzle as well.

**Comment by skapral** [comment]
> @skapral I left a comment; could you also write a test for these measurements? Or is it not possible yet because of the puzzles?

So far, I see no way of it to do in scope of 30 mins. budget. I created a separate issue for that: #107.

File: src/main/java/org/jpeek/App.java [hunk 2]
```diff
@@ -115,7 +115,9 @@ public void analyze() throws IOException {
         this.save(skeleton.toString(), "skeleton.xml");
         final Iterable<Report> reports = new ListOf<>(
             new Report(skeleton, "LCOM", this.params, 10.0d, -5.0d),
-            new Report(skeleton, "MMAC", this.params, 0.5d, 0.25d)
+            new Report(skeleton, "MMAC", this.params, 0.5d, 0.25d),
+            new Report(skeleton, "TCC",  this.params, 0.5d, 0.25d),
```
Comment (amihaiemil) [inline src/main/java/org/jpeek/App.java]: @skapral do not declare them here if they are not finished and tested. And please, leave more puzzles: 1 for writing some tests for these reports and 1 for declaring them here once all the other puzzles are finished :)

Comment (skapral) [reply src/main/java/org/jpeek/App.java]: @amihaiemil fixed

File: src/main/resources/org/jpeek/metrics/LCC.xsl (hunk 1 above)
Comment (skapral) [reply src/main/resources/org/jpeek/metrics/LCC.xsl]: @amihaiemil fixed for LCC and TCC descriptors.

**Comment by amihaiemil** [comment]
@rultor merge it pls

**Comment by amihaiemil** [comment]
@yegor256 ping :)

**Comment by amihaiemil** [comment]
@yegor256 hello? Can we merge this? :)

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Order was successfully finished: +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+105](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+8610](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) resigned from [#108](https://github.com/yegor256/jpeek/issues/108), since the job is not in scope anymore

**Comment by 0crat** [comment]
The job [#108](https://github.com/yegor256/jpeek/issues/108) is now out of scope

**Comment by 0crat** [comment]
This pull request [#108](https://github.com/yegor256/jpeek/issues/108) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) resigned from [#108](https://github.com/yegor256/jpeek/issues/108), since the job is not in scope anymore

**Comment by 0crat** [comment]
The job [#108](https://github.com/yegor256/jpeek/issues/108) is now out of scope

**Comment by 0crat** [comment]
Order was successfully finished: +30 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+180](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+8655](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
This pull request [#108](https://github.com/yegor256/jpeek/issues/108) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by 0crat** [comment]
Order was successfully finished: +15 points just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), total is [+1185](https://www.0crat.com/u/carlosmiranda)

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/jpeek/issues/108#issuecomment-366421791))

@g4s8 This is what I know about this job, as in [§32](http://datum.zerocracy.com/pages/policy.html#32):

  * The job [#108](https://github.com/yegor256/jpeek/issues/108) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C7JGJ00DP?q=%7Bjob%3A%27gh%3Ayegor256%2Fjpeek%23108%27%7D) (restricted area)

**Pull request merged**